rig/machine_control/sdram_file.py

```diff
@@ -0,0 +1,84 @@
+class SDRAMFile(object):
+    def __init__(self, machine_controller, x, y,
+                 start_address=0x60000000,
+                 end_address=0x6fffffff):
+        """Create a file-like view onto the SDRAM of a chip.
+
+        Parameters
+        ----------
+        machine_controller : :py:class:`~rig.machine_control.MachineController`
+            A communicator to handle transmitting and receiving packets from
+            the SpiNNaker machine.
+        x : int
+            The x co-ordinate of the chip to represent the SDRAM of.
+        y : int
+            The y co-ordinate of the chip to represent the SDRAM of.
+        start_address : int
+            Starting address of the SDRAM.
+        end_address : int
+            End address of the SDRAM.
+        """
+        # Store parameters
+        self._x = x
+        self._y = y
+        self._machine_controller = machine_controller
+        self._start_address = start_address
+        self._end_address = end_address
+
+        # Current offset from start address
+        self._offset = 0
+
+    def read(self, n_bytes):
+        """Read a number of bytes from the SDRAM.
+
+        Parameters
+        ----------
+        n_bytes : int
+            A number of bytes to read.
+
+        Returns
+        -------
+        :py:class:`bytes`
+            Data read from SpiNNaker as a bytestring.
+        """
+        assert self.address + n_bytes < self._end_address
```

Comment:
The standard behaviour of a python file is to produce no output when an EOF is reached and not to throw an exception.